File: tests/code/corpus/langs/csharp/Shapes.cs
using System;
using System.Collections.Generic;

namespace Geometry
{
    public interface IDrawable
    {
        void Draw();
    }

    public class Shape : IDrawable
    {
        public void Draw() { }

        public virtual double Area()
        {
            return 0.0;
        }
    }

    public class Circle : Shape
    {
        private readonly double radius;

        public Circle(double radius)
        {
            this.radius = radius;
        }

        public override double Area()
        {
            return Math.PI * radius * radius;
        }
    }

    public struct Point
    {
        public double X;
    }

    public enum Kind
    {
        Round,
        Square
    }

    public class Factory
    {
        public static Circle MakeCircle(double radius)
        {
            return new Circle(radius);
        }

        public static double TotalArea(List<Shape> shapes)
        {
            double total = 0.0;
            foreach (var shape in shapes)
            {
                total += shape.Area();
            }
            return total;
        }
    }
}

File: tests/code/corpus/langs/java/Shapes.java
package geometry;

import java.util.List;

public interface Drawable {
    void draw();
}

class Shape implements Drawable {
    public void draw() {}

    public double area() {
        return 0.0;
    }
}

class Circle extends Shape {
    private final double radius;

    Circle(double radius) {
        this.radius = radius;
    }

    public double area() {
        return Math.PI * radius * radius;
    }
}

enum Kind {
    ROUND,
    SQUARE
}

class Factory {
    static Circle makeCircle(double radius) {
        return new Circle(radius);
    }

    static double totalArea(List<Shape> shapes) {
        double total = 0.0;
        for (Shape shape : shapes) {
            total += shape.area();
        }
        return total;
    }
}
